lib/Sema/PreCheckExpr.cpp

```diff
@@ -602,8 +602,14 @@ Expr *TypeChecker::resolveDeclRefExpr(UnresolvedDeclRefExpr *UDRE,
       }
 
       TypoCorrectionResults corrections(Name, nameLoc);
-      TypeChecker::performTypoCorrection(DC, UDRE->getRefKind(), Type(),
-                                         lookupOptions, corrections);
+
+      // FIXME: Don't perform typo correction inside macro arguments, because it
```

Comment:
Argh, this is a tricky one. Maybe typo correction should avoid ever synthesizing declarations or expanding macros.